src/dataframe/dataframe.jl

```diff
@@ -530,6 +557,56 @@ function Base.setindex!(df::DataFrame,
     return df
 end
 
+function checkdimensions(new_df::AbstractDataFrame,
+                         row_inds::Any,
+                         col_inds::Any)
+    new_rows, new_cols = size(new_df)
+    # note that length(::Number) == 1
+    if (new_rows, new_cols) != (length(row_inds), length(col_inds))
+        msg = "Tried to assign $new_rows×$new_cols dataframe to " *
+              "$(length(row_inds))×$(length(col_inds)) destination"
+        throw(DimensionMismatch(msg))
+    end
+end
+
+function checknames(df::DataFrame,
+                  new_df::AbstractDataFrame,
+                  col_inds::AbstractVector{<:ColumnIndex})
+    new_names = _names(new_df)
+    subset = all(name -> haskey(df, name), new_names)
+    if !subset || !issetequal(index(df)[new_names], index(df)[col_inds])
+        msg = "Column names in inserted dataframe don't match the columns " *
+              "being inserted into"
+        throw(ArgumentError(msg))
+    end
+end
+
+function checkpreconditions(df::DataFrame,
```

Comment:
Better call `checknames` and `checkdimensions` directly, defining this helper method actually requires more lines of code.